src/ui/EditorNS/editor.cpp

```diff
@@ -148,9 +173,13 @@ namespace EditorNS
         else if(msg == "J_EVT_GOT_FOCUS")
             emit gotFocus();
         else if(msg == "J_EVT_CURRENT_LANGUAGE_CHANGED") {
-            QVariantMap map = data.toMap();
-            emit currentLanguageChanged(map.value("id").toString(),
-                                        map.value("name").toString());
+            QVariantMap lang = data.toMap();
+            QString id = lang.value("id").toString();
+            QString name = lang.value("name").toString();
+            if (!m_customIndentationMode) {
+                setIndentationMode(id);
+            }
+            emit currentLanguageChanged(id, name);
```

Comment:
For some reason the toolbar now shows the internal language name instead of the friendly one (e.g. "csharp" instead of "C#")